implement upload and template targets

trombik opened the pull request:
this PR introduces new targets to upload OVA files and create templates. supported upload place is `S3`.

`cloudstack-api` becomes a requirement.

Commit:
add .ackrc and update gitignore

Commit:
implement upload and template targets

Commit:
remove unusual ssh config

Commit:
remove environment-dependant variable

Commit:
implement ACL on objects

Commit:
syntax

Commit:
properly deep-merge the configs

Commit:
order

Commit:
wrong example

mitsururike commented on examples/Vagrantfile:
```diff
@@ -53,7 +52,7 @@ Vagrant.configure(2) do |config|
 
     # cloudstack-api listNetworks
-    cloudstack.network_name = "71000001146:10.102.36.0/23"
+    cloudstack.network_name = CHANGEME
```
if this is a working example, s/CHANGEME/"CHANGEME"/ to show that it must be a string.

mitsururike commented on config.yml:
```diff
@@ -0,0 +1,13 @@
+  # see cloudstack-api listZones
+  zoneid: XXXXXXXX-XXXX-XXXX-XXXXXXXXXXXXXXXXX
+  api_command: cloudstack-api
+  # see http://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html
```
this comment should be before `acl:` field.

Commit:
update examples

https://github.com/reallyenglish/box2cloudstack/pull/6#discussion_r144479155
https://github.com/reallyenglish/box2cloudstack/pull/6#discussion_r144481054

Commit:
use same box name throughout

mitsururike commented on config.yml:
```diff
@@ -0,0 +1,13 @@
+  upload: s3
+  # see cloudstack-api listZones
+  zoneid: XXXXXXXX-XXXX-XXXX-XXXXXXXXXXXXXXXXX
+  # see http://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html
```
this comment should be before `acl:` field of the `s3:` section as it explains what `public-read` is.

Commit:
wrong line
